src/main/java/org/myrobotlab/service/MqttBroker.java

```diff
@@ -0,0 +1,628 @@
+
+    } else if (msg instanceof InterceptUnsubscribeMessage) {
+      InterceptUnsubscribeMessage m = (InterceptUnsubscribeMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+      retMsg.setTopicName(m.getTopicFilter());
+    }
+    return retMsg;
+  }
+
+  protected String address = "0.0.0.0";
+
+  protected Boolean allow_anonymous = false;
+
+  protected Boolean allow_zero_byte_client_id = false;
+
+  protected Set<String> connectedClients = new HashSet<>();
+
+  /**
+   * determines if the broker will attempt to process any messages published on
+   * the api topics .. typically api/service & api/messages
+   */
+  boolean processApiMessages = true;
+
+  protected boolean listening = false;
+
+  transient final Server mqttBroker = new Server();
+
+  protected Integer mqttPort = 1883;
+
+  protected String mrlTopicApiPrefix = "api";
+
+  protected String messagesTopic = mrlTopicApiPrefix + "/" + CodecUtils.API_MESSAGES + "/";
+
+  protected String serviceTopic = mrlTopicApiPrefix + "/" + CodecUtils.API_SERVICE + "/";
+
+  protected Map<String, List<MRLListener>> notifyList = new HashMap<>();
+
+  protected String password = null;
+
+  protected String passwordFilePath = getResourceDir(MqttBroker.class, "password_file.conf");
+
+  protected String username = null;
+
+  protected Integer wsPort = 8080;
+
+  public MqttBroker(String n, String id) {
+    super(n, id);
+  }
+
+  @Override
+  public void connect(String uri) throws Exception {
+    // TODO Auto-generated method stub
+
+  }
+
+  public String getAddress() {
+    return address;
+  }
+
+  @Override
+  public List<String> getClientIds() {
+    return Runtime.getInstance().getConnectionUuids(getName());
+  }
+
+  @Override
+  public Map<String, Connection> getClients() {
+    return Runtime.getInstance().getConnections(getName());
+  }
+
+  @Override
+  public Message getDescribeMsg(String connId) {
+    return Runtime.getInstance().getDescribeMsg(connId);
+  }
+
+  @Override
+  public String getID() {
+    return getName();
+  }
+
+  @Override
+  public Class<?>[] getInterceptedMessageTypes() {
+    Class<?>[] ALL_MESSAGE_TYPES = { InterceptConnectMessage.class, InterceptDisconnectMessage.class, InterceptConnectionLostMessage.class, InterceptPublishMessage.class,
+        InterceptSubscribeMessage.class, InterceptUnsubscribeMessage.class, InterceptAcknowledgedMessage.class };
+
+    return ALL_MESSAGE_TYPES;
+  }
+
+  public Integer getMqttPort() {
+    return mqttPort;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public Integer getWsPort() {
+    return wsPort;
+  }
+
+  public boolean isLocal(Message msg) {
+    return Runtime.getInstance().isLocal(msg);
+  }
+
+  public void listen() {
+    listen(null, null, null, null, null, null);
+  }
+
+  public void listen(int mqttPort) {
+    this.mqttPort = mqttPort;
+    listen();

```

Comment:
you should call the max complexity method.  not listen() ...    that's the least complexity method!
listen(null,  inMqttPort, null, null, null, null)

Reply:
backwards ...

will fix

Reply:
fixed